crates/primitives/src/proof.rs

```diff
@@ -0,0 +1,38 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+
+use crate::buf::Buf32;
+
+#[derive(Clone, Debug, Copy, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum RollupVerifyingKey {
+    // Verifying Key used to verify proof created using SP1
+    SP1VerifyingKey(Buf32),
+    // Verifying Key used to verify proof created using Risc0
+    Risc0VerifyingKey(Buf32),
```

Comment:
While we're here can we just do `#[serde(rename = "sp1")]` and `#[serde(rename = "r0")]` so we don't have the annoying `s_p1_verifiying_key` thing.